modular/blocksyncer/blocksyncer_options.go

```diff
@@ -249,50 +231,65 @@ func (b *BlockSyncerModular) getLatestBlockHeight(ctx context.Context) {
 
 func (b *BlockSyncerModular) quickFetchBlockData(startHeight uint64) {
 	count := uint64(b.config.Parser.Workers)
-	for cycle := uint64(0); ; cycle++ {
+	cycle := uint64(0)
+	for {
 		latestBlockHeightAny := Cast(b.parserCtx.Indexer).GetLatestBlockHeight().Load()
 		latestBlockHeight := latestBlockHeightAny.(int64)
 		if latestBlockHeight < int64(count*(cycle+1)+startHeight-1) {
 			log.Infof("quick fetch ended latestBlockHeight: %d", latestBlockHeight)
 			Cast(b.parserCtx.Indexer).GetCatchUpFlag().Store(int64(count*cycle + startHeight - 1))
 			break
 		}
-		wg := &sync.WaitGroup{}
-		wg.Add(int(count))
-		for i := uint64(0); i < count; i++ {
-			go func(idx, c uint64) {
-				defer wg.Done()
-				height := idx + count*c + startHeight
-				if height > uint64(latestBlockHeight) {
-					return
+		processedHeight, ok := <-Cast(b.parserCtx.Indexer).ProcessedQueue
+		if !ok {
+			log.Warnf("ProcessedQueue is closed")
+			return
+		}
+		log.Infof("processedHeight:%d, will process height:%d", processedHeight, count*cycle+startHeight)
+		if processedHeight != 0 && count*cycle+startHeight-processedHeight > 4*count {
```

Comment:
I think '4' here means block cache capacity for each worker, define this as a const value and name it like 'BlockCacheCapacity' or other meaningful names